core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -732,6 +732,14 @@ class KafkaApis(val requestChannel: RequestChannel,
         response.responseData.asScala.foreach { case (topicPartition, data) =>
           // record the bytes out metrics only when the response is being sent
           brokerTopicStats.updateBytesOut(topicPartition.topic, fetchRequest.isFromFollower, data.records.sizeInBytes)
+          // record any outgoing reassignment rate if the fetch request came from an addingReplica of the partition
```

Comment:
On a second thought it would be more straightforward to add an `isAddingReplica` method to ReplicaManager which would do this functionality. Tried out FetchPartitionData but it is more cumbersome.